tests/unit/api/test_object_communicator.cc

```diff
@@ -0,0 +1,219 @@
+//@HEADER
+*/
+
+#include <gtest/gtest.h>
+
+#include <vt-tv/api/object_communicator.h>
+
+#include <string>
+#include <filesystem>
+#include <iostream>
+#include <variant>
+#include <set>
+
+#include "basic_serializer.h"
+
+namespace vt::tv::tests::unit::api {
+
+/**
+ * Provides unit tests for the vt::tv::api::ObjectCommunicator class
+ */
+class ObjectCommunicatorTest :public ::testing::Test {
+  public:
+    ObjectCommunicator comm_0 = ObjectCommunicator(0);
+
+    ObjectCommunicator comm_1 = ObjectCommunicator(
+        1,
+        // recv
+        {
+            { 1, 25.5 }, { 2, 12.0 }
+        },
+        // sent
+        {
+            { 1, 29.5 },  { 2, 10.2 },  { 3, 10.3 }
+        }
+    );
+};
+
+/**
+ * Test empty ObjectCommunicator initial state
+ */
+TEST_F(ObjectCommunicatorTest, test_empty_communicator_getters) {
+  // Assertions for comm_0
+  EXPECT_EQ(comm_0.getObjectId(), 0);
+  EXPECT_EQ(comm_0.getMaxVolume(), 0.0);
+  EXPECT_EQ(comm_0.getReceived().size(), 0);
+  EXPECT_EQ(comm_0.getTotalReceivedVolume(), 0.0);
+  EXPECT_EQ(comm_0.getSent().size(), 0);
+  EXPECT_EQ(comm_0.getTotalSentVolume(), 0.0);
+}
+
+/**
+ * Test ObjectCommunicator initial state
+ */
+TEST_F(ObjectCommunicatorTest, test_sample_communicator_and_getters) {
+  // Assertions for comm_1
+  EXPECT_EQ(comm_1.getObjectId(), 1);
+  EXPECT_EQ(comm_1.getMaxVolume(), 29.5);
+  EXPECT_EQ(comm_1.getReceived().size(), 2);
+  EXPECT_EQ(comm_1.getTotalReceivedVolume(), 37.5);
+  EXPECT_EQ(comm_1.getSent().size(), 3);
+  EXPECT_EQ(comm_1.getTotalSentVolume(), 50.0);
+}
+
+/**
+ * Test ObjectCommunicator initial state
+ */
+TEST_F(ObjectCommunicatorTest, test_initial_state) {
+  // Assertions for comm_0
+  EXPECT_EQ(comm_0.getObjectId(), 0);
+  EXPECT_EQ(comm_0.getMaxVolume(), 0.0);
+  EXPECT_EQ(comm_0.getReceived().size(), 0);
+  EXPECT_EQ(comm_0.getTotalReceivedVolume(), 0.0);
+  EXPECT_EQ(comm_0.getSent().size(), 0);
+  EXPECT_EQ(comm_0.getTotalSentVolume(), 0.0);
+
+  // Assertions for comm_1
+  EXPECT_EQ(comm_1.getObjectId(), 1);
+  EXPECT_EQ(comm_1.getMaxVolume(), 29.5);
+  EXPECT_EQ(comm_1.getReceived().size(), 2);
+  EXPECT_EQ(comm_1.getTotalReceivedVolume(), 37.5);
+  EXPECT_EQ(comm_1.getSent().size(), 3);
+  EXPECT_EQ(comm_1.getTotalSentVolume(), 50.0);
+}
+
+/**
+ * Test ObjectCommunicator initial state
+ */
+TEST_F(ObjectCommunicatorTest, test_summarize_communications_count_empty_communicator) {
+  // std::make_pair(w_sent, w_recv);
+  auto summary = comm_0.summarize();
+  EXPECT_EQ(summary.first.size(), 0);
+  EXPECT_EQ(summary.second.size(), 0);
+}
+
+/**
+ * Test ObjectCommunicator initial state
+ */
+TEST_F(ObjectCommunicatorTest, test_summarize_communications_count) {
+  GTEST_SKIP() << "ObjectCommunicator::summarize seems in WIP and is never called";
+
+  // std::make_pair(w_sent, w_recv);
+  auto summary = comm_0.summarize();
+  EXPECT_EQ(summary.first.size(), 3);
+  EXPECT_EQ(summary.second.size(), 2);
```

Comment:
Since this test isn't being run, I think we can just delete this stuff

Reply:
In fact there is an underlying question ! It seems that the `summarize` method is never called from the source. Furthermore this method implementation (class `ObjectCommunicator`) always returns a pair of empty vectors due to commented code in `summarizeUnidirectional` method. Is there any reason to keep this in sources ? Until we determine that point I suggest to let this skipped test. If it must be kept I think @pierrepebay you can create a follow up issue

Reply:
`summarizeUnidirectional` and `summarize` are something I started adding because it was in `LBAF`, but never completely implemented. As you said it's not used anywhere, and I don't think it will as it isn't particularly useful for `vt-tv`. I think we can remove both functions, and those tests accordingly.

Reply:
I've removed these functions and their tests from vt-tv